dispatch/static/manager/src/js/actions/ArticlesActions.js

```diff
@@ -1,210 +1,59 @@
-          payload: payload
-        })
-      })
-      .catch(error => {
-        dispatch({
-          type: `${types.PUBLISH_ARTICLE}_REJECTED`,
-          payload: error
-        })
-      })
-  }
-}
-
-export function unpublishArticle(token, articleId) {
-  return {
-    type: types.UNPUBLISH_ARTICLE,
-    payload: DispatchAPI.articles.unpublishArticle(token, articleId)
-      .then( json => normalize(json, articleSchema) )
-  }
-}
+    data.author_ids = data.authors
 
-export function previewArticle(token, articleId, data) {
+    data.tag_ids = data.tags
 
-  data = preparePayload(data)
+    data.topic_id = data.topic
 
-  return function(dispatch) {
-    dispatch({ type: `${types.SAVE_ARTICLE}_PENDING` })
+    data.template_id = data.template
+    delete data.template
 
-    DispatchAPI.articles.saveArticle(token, articleId, data)
-      .then(json => {
-        dispatch({
-          type: `${types.SAVE_ARTICLE}_FULFILLED`,
-          payload: normalize(json, articleSchema)
-        })
+    if (data.featured_image) {
+      data.featured_image.image_id = data.featured_image.image
+    }
 
-        // Open article in new browser tab
-        window.open(json.url, `_dispatch-preview-${json.id}`)
-      })
-      .catch(error => {
-        dispatch({
-          type: `${types.SAVE_ARTICLE}_REJECTED`,
-          payload: error
-        })
-      })
-
-  }
-}
-
-export function toggleArticle(articleId) {
-  return {
-    type: types.TOGGLE_ARTICLE,
-    id: articleId
+    return data
   }
-}
 
-export function toggleAllArticles(articleIds) {
-  return {
-    type: types.TOGGLE_ALL_ARTICLES,
-    ids: articleIds
-  }
-}
+  search(section, query) {
+    var queryObj = {}
 
-export function clearSelectedArticles() {
-  return {
-    type: types.CLEAR_SELECTED_ARTICLES
-  }
-}
+    if (query) {
+      queryObj.q = query
+    }
 
-export function deleteArticles(token, articleIds) {
-  return function(dispatch) {
-    dispatch({ type: `${types.DELETE_ARTICLES}_PENDING` })
-
-    Promise.all(
-      articleIds.map(articleId => DispatchAPI.articles.deleteArticle(token, articleId))
-    )
-    .then(() => {
-      dispatch({
-        type: `${types.DELETE_ARTICLES}_FULFILLED`,
-        payload: articleIds
-      })
-    })
-    .catch(error => {
-      dispatch({
-        type: `${types.DELETE_ARTICLES}_REJECTED`,
-        payload: error
-      })
-    })
+    if (section) {
+      queryObj.section = section
+    }
 
+    return function(dispatch) {
+      dispatch(push({ pathname: '/articles/', query: queryObj }))
+    }
   }
-}
 
-export function clearArticles() {
-  return {
-    type: types.CLEAR_ARTICLES
-  }
 }
 
-export function searchArticles(section, query) {
-
-  var queryObj = {}
-
-  if (query) {
-    queryObj.q = query
-  }
-
-  if (section) {
-    queryObj.section = section
-  }
-
-  return function(dispatch) {
-    dispatch(push({ pathname: '/articles/', query: queryObj }))
-  }
-}
+export default new ArticlesActions(
```

Comment:
Do you think it would be cleaner to export an instance of a named class expression instead?
```js
export default new class ArticlesActions extends PublishableActions {
  prepareData(data) {
    ...
  }
  search(section, query) {
    ...
  }
}(
  types.ARTICLES,
  DispatchAPI.articles,
  articleSchema
)
```